src/activation.jl

```diff
@@ -204,25 +209,29 @@ logcosh(x::Real) = x + softplus(-2x) - log(oftype(x, 2))
 Self Regularized Non-Monotonic Neural Activation Function.
 See [Mish: A Self Regularized Non-Monotonic Neural Activation Function](https://arxiv.org/abs/1908.08681).
 """
-mish(x::Real) = x * tanh(softplus(x))
+mish(x::RealOrFloatType) = x * tanh(softplus(x))
 
 """
     tanhshrink(x) = x - tanh(x)
 
 See [Tanhshrink Activation Function](https://www.gabormelli.com/RKB/Tanhshrink_Activation_Function).
 """
-tanhshrink(x::Real) = x - tanh(x)
+tanhshrink(x::RealOrFloatType) = x - tanh(x)
 
 """
     softshrink(x, λ=0.5) =
         (x ≥ λ ? x - λ : (-λ ≥ x ? x + λ : 0))
 
 See [Softshrink Activation Function](https://www.gabormelli.com/RKB/Softshrink_Activation_Function).
 """
-softshrink(x::Real, λ = oftype(x/1, 0.5)) = min(max(zero(x), x - λ), x + λ)
+softshrink(x::RealOrFloatType, λ = oftype(x/1, 0.5)) = min(max(zero(x), x - λ), x + λ)
 
 # Provide an informative error message if activation functions are called with an array
 for f in (:σ, :hardσ, :logσ, :hardtanh, :relu, :leakyrelu, :relu6, :rrelu, :elu, :gelu, :swish, :lisht, :selu, :celu, :trelu, :softsign, :softplus, :logcosh, :mish, :tanhshrink, :softshrink)
     @eval $(f)(x::AbstractArray, args...) =
       error("Use broadcasting (`", $(string(f)), ".(x)`) to apply activation functions to arrays.")
 end
+
+for f in (:σ, :tanh)
+    @eval Base.broadcasted(::typeof($f), x::AbstractArray{T, N}) where {T <: Union{Float64, Float32}, N} = vmap($f, x)
```

Comment:
we can extend this to all activations, right?
Also, maybe better play and restrict the signature to `Array{T, N}`, since my understanding is that `vmap` doesn't support e.g. StaticArrays yet

Reply:
We cannot extend this to all activations because the adjoint of `vmap` is not defined. `σ` and `tanh` are special because their broadcasting adjoints are explicitly definded in Zygote. See https://github.com/FluxML/Zygote.jl/blob/40db80ce0d9344d0a30d011cacd359630809db4d/src/lib/broadcast.jl#L99.

We should restrict the signature to `Array{T, N}` and also `SubArray`.